File: correctness-as241.js
#!/usr/bin/env node
var as241 = require("./as241.js")

function* range(start,end,increment) {
  comp = start < end ? (x,y) => x < y : (x,y) => x > y
  for (let i = start; comp(i,end); i += increment) {
    yield i
  }
}

values = [
  ...Array.from(range(600,0,-1)).map((x) => "1e-" + x),
  ...Array.from(range(0.01,1,0.01))
]

console.log("value\tninv")
for (let v of values) {
  r = as241.ninv(v)
  console.log(v + "\t" + String(r))
}

File: correctness-as241.py
#!/usr/bin/env python3
import numpy as np
from as241 import ninv

values = ["1e-" + str(x) for x in range(600,0,-1)] + list(map(str,np.arange(0.01,1,0.01)))

print("value\tninv")
for v in values:
  r = ninv(np.float128(v))
  print(v,"\t",r)

